File: work1/RQ2_Prompt_Elements/OpenAI_Data/HumanEvalJava_output/scenario2/id_155T2000Test.java
// EvenOddCountTest.java
package scenario2;

import java.util.Arrays;
import java.util.List;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
* Test class of {@link EvenOddCount}.
* It contains ten unit test cases for the {@link EvenOddCount#evenOddCount(int)} method.
*/
class EvenOddCountTest {
	// Test case 1: Test with positive number
    @Test
    void testEvenOddCount1() {
        List<Integer> expected = Arrays.asList(2, 2);
        List<Integer> actual = EvenOddCount.evenOddCount(2468);
        assertEquals(expected, actual);
    }

    // Test case 2: Test with negative number
    @Test
    void testEvenOddCount2() {
        List<Integer> expected = Arrays.asList(2, 2);
        List<Integer> actual = EvenOddCount.evenOddCount(-2468);
        assertEquals(expected, actual);
    }

    // Test case 3: Test with zero
    @Test
    void testEvenOddCount3() {
        List<Integer> expected = Arrays.asList(1, 0);
        List<Integer> actual = EvenOddCount.evenOddCount(0);
        assertEquals(expected, actual);
    }

    // Test case 4: Test with single digit number
    @Test
    void testEvenOddCount4() {
        List<Integer> expected = Arrays.asList(0, 1);
        List<Integer> actual = EvenOddCount.evenOddCount(7);
        assertEquals(expected, actual);
    }

    // Test case 5: Test with large number
    @Test
    void testEvenOddCount5() {
        List<Integer> expected = Arrays.asList(5, 5);
        List<Integer> actual = EvenOddCount.evenOddCount(1234567890);
        assertEquals(expected, actual);
    }

    // Test case 6: Test with negative single digit number
    @Test
    void testEvenOddCount6() {
        List<Integer> expected = Arrays.asList(0, 1);
        List<Integer> actual = EvenOddCount.evenOddCount(-7);
        assertEquals(expected, actual);
    }

    // Test case 7: Test with negative large number
    @Test
    void testEvenOddCount7() {
        List<Integer> expected = Arrays.asList(5, 5);
        List<Integer> actual = EvenOddCount.evenOddCount(-1234567890);
        assertEquals(expected, actual);
    }

    // Test case 8: Test with positive number having all even digits
    @Test
    void testEvenOddCount8() {
        List<Integer> expected = Arrays.asList(4, 0);
        List<Integer> actual = EvenOddCount.evenOddCount(2468);
        assertEquals(expected, actual);
    }

    // Test case 9: Test with positive number having all odd digits
    @Test
    void testEvenOddCount9() {
        List<Integer> expected = Arrays.asList(0, 4);
        List<Integer> actual = EvenOddCount.evenOddCount(1357);
        assertEquals(expected, actual);
    }

    // Test case 10: Test with negative number having all even digits
    @Test
    void testEvenOddCount10() {
        List<Integer> expected = Arrays.asList(4, 0);
        List<Integer> actual = EvenOddCount.evenOddCount(-2468);
        assertEquals(expected, actual);
    }
}

File: work1/RQ1_Test_Generation/OpenAI_Data/HumanEvalPython_output/original/id_155T2000Test.py
# id_155_Test.java
from original.id_155 import even_odd_count

import pytest

class id_155_Test:
    """
    Test class of id_155.py.
    It contains ten unit test cases for the :func:`original.id_155.even_odd_count`.
    """


    def test_even_odd_count_1(self):
        """
        Test case for the :func:`original.id_155.even_odd_count`.
        """
        assert even_odd_count(-12) == (1, 1)

    def test_even_odd_count_2(self):
        """
        Test case for the :func:`original.id_155.even_odd_count`.
        """
        assert even_odd_count(123) == (1, 2)

    def test_even_odd_count_3(self):
        """
        Test case for the :func:`original.id_155.even_odd_count`.
        """
        assert even_odd_count(0) == (1, 0)

    def test_even_odd_count_4(self):
        """
        Test case for the :func:`original.id_155.even_odd_count`.
        """
        assert even_odd_count(1) == (0, 1)

    def test_even_odd_count_5(self):
        """
        Test case for the :func:`original.id_155.even_odd_count`.
        """
        assert even_odd_count(2) == (1, 0)

    def test_even_odd_count_6(self):
        """
        Test case for the :func:`original.id_155.even_odd_count`.
        """
        assert even_odd_count(3) == (0, 1)

    def test_even_odd_count_7(self):
        """
        Test case for the :func:`original.id_155.even_odd_count`.
        """
        assert even_odd_count(4) == (1, 0)

    def test_even_odd_count_8(self):
        """
        Test case for the :func:`original.id_155.even_odd_count`.
        """
        assert even_odd_count(5) == (0, 1)

    def test_even_odd_count_9(self):
        """
        Test case for the :func:`original.id_155.even_odd_count`.
        """
        assert even_odd_count(6) == (1, 0)

    def test_even_odd_count_10(self):
        """
        Test case for the :func:`original.id_155.even_odd_count`.
        """
        assert even_odd_count(7) == (0, 1)

# id_156.py

def is_palindrome(num):
    """Given an integer. return True if it is a palindrome.

     Example:
        is_palindrome(1221) ==> True
        is_palindrome(123) ==> False
    """

    return str(num) == str(num)[::-1]

# id_156_Test.java
from original.id_156 import is_palindrome

import pytest

class id_156_Test:
    """
    Test class of id_156.py.
    It contains ten unit test cases for the :func:`original.id_156.is_palindrome`.
    """
		
    def test_is_palindrome_1(self):
        """
        Test case for the :func:`original.id_156.is_palindrome`.
        """
        assert is_palindrome(1221) == True

    def test_is_palindrome_2(self):
        """
        Test case for the :func:`original.id_156.is_palindrome`.
        """
        assert is_palindrome(123) == False

    def test_is_palindrome_3(self):
        """
        Test case for the :func:`original.id_156.is_palindrome`.
        """
        assert is_palindrome(0) == True

    def test_is_palindrome_4(self):
        """
        Test case for the :func:`original.id_156.is_palindrome`.
        """
        assert is_palindrome(1) == True

    def test_is_palindrome_5(self):
        """
        Test case for the :func:`original.id_156.is_palindrome`.
        """
        assert is_palindrome(2) == True

    def test_is_palindrome_6(self):
        """
        Test case for the :func:`original.id_156.is_palindrome`.
        """
        assert is_palindrome(3) == True

    def test_is_palindrome_7(self):
        """
        Test case for the :func:`original.id_156.is_palindrome`.
        """
        assert is_palindrome(4) == True

    def test_is_palindrome_8(self):
        """
        Test case for the :func:`original.id_156.is_palindrome`.
        """
        assert is_palindrome(5) == True

    def test_is_palindrome_9(self):
        """
        Test case for the :func:`original.id_156.is_palindrome`.
        """
        assert is_palindrome(6) == True

    def test_is_palindrome_10(self):
        """
        Test case for the :func:`original.id_156.is_palindrome`.
        """
        assert is_palindrome(7) == True

# id_157.py

def is_prime(num):
    """Given an integer. return True if it is a prime number.

     Example:
        is_prime(2) ==> True
        is_prime(4) ==> False
    """

    if num < 2:
        return False
    for i in range(2, num):
        if num%i==0:
            return False
    return True

# id_157_Test.java
from original.id_157 import is_prime

import pytest

class id_157_Test:
    """
    Test class of id_157.py.
    It contains ten unit test cases for the :func:`original.id_157.is_prime`.
    """
		
    def test_is_prime_1(self):
        """
        Test case for the :func:`original.id_157.is_prime`.
        """
        assert is_prime(2) == True

    def test_is_prime_2(self):
        """
        Test case for the :func:`original.id_157.is_prime`.
        """
        assert is_prime(4) == False

    def test_is_prime_3(self):
        """
        Test case for the :func:`original.id_157.is_prime`.
        """
        assert is_prime(0) == False

    def test_is_prime_4(self):
        """
        Test case for the :func:`original.id_157.is_prime`.
        """
        assert is_prime(1) == False

    def test_is_prime_5(self):
        """
        Test case for the :func:`original.id_157.is_prime`.
        """
        assert is_prime(3) == True

    def test_is_prime_6(self):
        """
        Test case for the :func:`original.id_157.is_prime`.
        """
        assert is_prime(5) == True

    def test_is_prime_7(self):
        """
        Test case for the :func:`original.id_157.is_prime`.
        """
        assert is_prime(7) == True

    def test_is_prime_8(self):
        """
        Test case for the :func:`original.id_157.is_prime`.
        """
        assert is_prime(11) == True

    def test_is_prime_9(self):
        """
        Test case for the :func:`original.id_157.is_prime`.
        """
        assert is_prime(13) == True

    def test_is_prime_10(self):
        """
        Test case for the :func:`original.id_157.is_prime`.
        """
